### DISCUSSION TITLE
Retrieve full taxa names

### DISCUSSION THREAD

**Discussion by BlueK77** [FrederickHuangLin/ANCOMBC#169] [Q&A]
When I use [tax_level = "Genus"], I obtain several g__uncultured which are labeled sequentially (_1, _2, _3, etc..). It is tedious to find out who is who manually and quite laborious if I want to filter the abundance of just those genus significant different (for example, for correlation porpoise).  Thus I was wondering if it possible or how would be possible to retrieve the full taxa names in a more elegant and efficient way. Thanx for suggestions

**Comment by FrederickHuangLin** [comment]
Hi @BlueK77,

The easiest way to achieve this is by creating a new column called "Genus" that combines the full taxon names. However, this step cannot be performed within the ANCOMBC functions themselves. Instead, it should be done during the data preprocessing stage, before running the analysis.

Best,
Huang
